test/test_transformer.cc

```diff
@@ -518,3 +518,211 @@ TEST(Transformer, Codegen) {
+  // clang-format off
+  auto matcher = band(
+    [&node] (isl::schedule_node n) {
+      if(isl_schedule_node_band_n_member(n.get()) < 3)
+        return false;
+      else { node = n; return true; }
+    },
+    leaf());
+  // clang-format on
+
+  bool matched = false;
+  if(ScheduleNodeMatcher::isMatching(matcher, root.child(0))) {
+    matched = true;
+  } 
+
+  isl::union_map reads = scop.reads.curry();
+  isl::union_map writes = scop.mustWrites.curry();
+
+  auto _i = placeholder(ctx);
+  auto _j = placeholder(ctx);
+  auto _k = placeholder(ctx);
+  auto _ii = placeholder(ctx);
+  auto _jj = placeholder(ctx);
+
+  auto _A = arrayPlaceholder();
+  auto _B = arrayPlaceholder();
+  auto _C = arrayPlaceholder();
+
+  auto psRead = allOf(access(_A, _i, _j), 
+		      access(_B, _i, _k),
+		      access(_C, _k, _j));
+  auto matchedReads = 
+    match(reads, psRead);
+  //std::cout << "matched Reads " << matchedReads.size() << std::endl;
+  auto psWrite = allOf(access(_A, _ii, _jj));
+  auto matchedWrites =
+    match(writes, psWrite);
+  //std::cout << "matched Writes " << matchedWrites.size() << std::endl;
+
+  // tmp[j][i] = alpha * A[i][k] * B[k][j] + tmp[i][j]
+  // pass the checks. this is because we do not link
+  // read and write at the moment. Placeholder are _not_
+  // reused between different calls to allOf. We can overcome
+  // this inspecting the placeholder for the write and the read.
+  // They should be equal. 
+ 
+  matched = matched && (matchedWrites.size() == 1);
+  matched = matched && (matchedReads.size() == 1);
+  EXPECT_TRUE(matched);
+
+  // check index for read and write are equal
+  matched = matched && 
+    matchedWrites[0][_ii].payload().inputDimPos_ == matchedReads[0][_i].payload().inputDimPos_;
+  matched = matched &&
+    matchedWrites[0][_jj].payload().inputDimPos_ == matchedReads[0][_j].payload().inputDimPos_;
+  EXPECT_TRUE(matched);
+
+  // D[i][j] = alpha * A[i][k] * B[k][j] + tmp[i][j]
+  // pass the test. We may want to apply the same check
+  // as before also for the accessed array.
+
+  // step 1. (not implemented, just assumed)
+  // we assume the indexes are already in
+  // the "good" order. (i,j,k).
+  // Line 1285 to 1291 in ScheduleOptimizer.cpp
+
+  // step 2. create macro-kernel
+  // line 1294 in ScheduleOptimizer.cpp
+  // For the micro and macro kernels we assume
+  // given values for the tile size.
+  using namespace builders;
+  // tiling - Mc=96, Nc=2048, Kc=512
+  // set tile values manually
+  int dimOutNum = isl_schedule_node_band_n_member(node.get());
+  std::vector<int> tileSizes(dimOutNum);
+  tileSizes[dimOutNum - 3] = /*96*/32;
+  tileSizes[dimOutNum - 2] = /*2048*/32;
+  tileSizes[dimOutNum - 1] = /*512*/32;
+  // tile node and get partial schedule 
+  isl::schedule_node tiledNode = tile(node, tileSizes);
+  auto tileSchedule = tiledNode.band_get_partial_schedule();
+  auto pointSchedule = tiledNode.child(0).band_get_partial_schedule();
+  // id for mark nodes.
```

Comment:
mark nodes are useless, polly does them for annotation, but we don't need them in the test IMO

Reply:
Ok I can remove them I think with mark not it is easy to check what's going on while you are transforming.

Reply:
Keep them if you want but use them in checks then :)